toolkit.py

```diff
@@ -0,0 +1,196 @@
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        hlines = cv.reduce(image[:, :, 1], 0,
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        vlines = map(lambda *row: list(row), *vlines)
+        y = np.diff(vlines[0])
+        x = np.diff(hlines[0])
+        return (self.__get_extremes(x), self.__get_extremes(y))
+
+    def get_table_metadata(self, image):
+        """
+        Function distorts image with median blur to
+        remove noise from image and unify colors a bit
+
+        Args:
+            image(numpy.ndarray): Loaded image
+
+        Returns:
+            Board's corner coordinates
+        """
+        process_image = cv.medianBlur(image, 5)
+        return self.get_board_boundaries(process_image)
+
+    def __get_file_name(self, path):
+        """
+        Args:
+            path(string): Path to file
+        Returns:
+            string: File name extracted from path
+        """
+        head, tail = ntpath.split(path)
+        name = tail or ntpath.basename(head)
+        return name.split('.')[0]
+
+    def write_video_metadata(self, video, meta_file):
+        """
+        Sets up a video capture and loops through all of the
+        frames. Pulls frame metadata from every frame and stores
+        it into an array. After we are done with a video we flush these
+        values to a file
+
+        Args:
+            video(string): Name of video file to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        vid = cv.VideoCapture(video)
+        if vid is None:
+            print("Couldn't load video {}".format(video), file=sys.stderr)
+            return
+        frames = []
+        while(vid.isOpened()):
+            ret, frame = vid.read()
+            if not ret:
+                break
+            (x, y) = self.get_table_metadata(frame)
+            frames.append((x, y))
+        with meta_file as csvfile:
+            for i, frame in enumerate(frames):
+                csv_writer = csv.writer(csvfile, lineterminator='\n')
+                csv_writer.writerow([x[0], x[1], y[0], y[1]])
+        vid.release()
+
+    def write_image_metadata(self, image, meta_file):
+        """
+        Reads given image and gets his metadata which
+        are written to a file
+
+        Args:
+            image(string): Name of image to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        im = cv.imread(image, cv.CV_LOAD_IMAGE_COLOR)
+        if im is None:
+            print("Couldn't read image file {}".format(image), file=sys.stderr)
+            return
+        (x, y) = self.get_table_metadata(im)
+        with meta_file as csvfile:
+            csv_writer = csv.writer(csvfile, lineterminator='\n')
+            csv_writer.writerow([x[0], x[1], y[0], y[1]])
+
+    video_extension_list = ("mkv", "wmv", "avi", "mp4")
```

Comment:
Please do not put variables in between function definitions. I personally think that they belong to the top of the class definition but I realize. That pep might not explicitly mention that.